### PR TITLE
Container image replacement if current image ref. uses digest

### LINKED ISSUE #14604: KubernetesManifest task does not replace container image when referenced by digest
## Required Information

**Question, Bug, or Feature?**  
*Type*: bug

**Enter Task Name**: KubernetesManifestV0

## Environment

- Server - Azure Pipelines
- Agent - Hosted
    
## Issue Description

When an existing Kubernetes deployment contains a container image reference using the digest format, the container image reference is not updated by the task.

Given a deployment spec:
```
kind: Deployment
apiVersion: apps/v1
spec:
  replicas: 1
  selector:
    matchLabels:
      app: foo
  template:
    spec:
      containers:
        - name: foo-image
          image: >-
            bar.azurecr.io/foo-image@sha256:4b523e3b94d90dfaaa7e075d52f43be8407a32e68203a1678be659132f4e1393
```
And the step:

```
- task: KubernetesManifest@0
  inputs:
    action: deploy
    namespace: ${{ parameters.environmentName }}
    manifests: |
      $(deployFile)
      $(svcFile)
    imagePullSecrets: |
      devops-pullsecret-${{ parameters.environmentName }}
    containers: |
       bar.azurecr.io/foo-image:20210311.1
```

The container reference is not being replaced, as expected.

### PR DESCRIPTION
**Pull request by vidarkongsli**
**Task name**: KubernetesManifestV0

**Description**: Fixes #14604 

**Documentation changes required:** (Y/N) N

**Added unit tests:** (Y/N) Y

**Attached related issue:** (Y/N) Y

**Checklist**:
- [x] Task version was bumped - please check [instruction](https://github.com/microsoft/azure-pipelines-tasks/tree/master/docs/taskversionbumping.md) how to do it
- [ ] Checked that applied changes work as expected

### CODE REVIEW

File: Tasks/KubernetesManifestV0/Tests/L0.ts [hunk 1]
```diff
@@ -436,15 +436,15 @@ describe('Kubernetes Manifests Suite', function () {
         const deploymentFile = fs.readFileSync(testFile).toString();
         const deploymentObject = yaml.load(deploymentFile);
 
-        updateImageDetails(deploymentObject, ['nginx:42', 'mysql:8.0', 'imagewithhyphen:1', 'myacr.azurecr.io/myimage:1', 'myimagewithcomment:1', 'nginx-init:42.1', 'myacr.azurecr.io/folder/image-a:1', 'myacr.azurecr.io/folder/image-b:2']);
+        updateImageDetails(deploymentObject, ['nginx:42', 'mysql:8.0', 'imagewithhyphen:1', 'myacr.azurecr.io/myimage:1', 'myimagewithcomment:1', 'nginx-init:42.1', 'myacr.azurecr.io/folder/image-a:1', 'myacr.azurecr.io/folder/image-b:2','mcr.microsoft.com/dotnet/core/sdk:1']);
```
Comment (ds-ms) [inline Tasks/KubernetesManifestV0/Tests/L0.ts:439]: Could you add a test for the other way round too? where there's a sha is your input, but a normal tag exists in the deployment.yml

Comment (vidarkongsli) [reply Tasks/KubernetesManifestV0/Tests/L0.ts]: Done.

**Review by ds-ms — APPROVED** [review]
LGTM 👍🏻

**Comment by ds-ms** [comment]
I'll merge this once the ci builds succeed.

**Comment by ds-ms** [comment]
@vidarkongsli if this needs to immediately fixed; I'll get this rollout started next week. 😄

**Comment by vidarkongsli** [comment]
@ds-ms Once I found the problem, I created a workaround for my case. So there is no immediate urgency for me.

**Pull request merged**